lib/elixir/test/elixir/kernel/with_test.exs

```diff
@@ -50,6 +54,17 @@ defmodule Kernel.WithTest do
     end
   end
 
+  test "with clause error" do
+    assert_raise WithClauseError, fn ->
```

Comment:
I'd be better to match on error message in this assertion:
```elixir
assert_raise WithClauseError, "no with clause matching: 41", fn ->
```
and remove next test.